react/game/src/app/screens/Game/layout.js

```diff
@@ -0,0 +1,42 @@
+import React from 'react';
+import { arrayOf, bool, number, string, func, shape } from 'prop-types';
+
+import styles from './styles.scss';
+import Board from './components/Board';
+
+function GameLayout({ history, stepNumber, checkStatus, movement, winner, xIsNext, handleClick }) {
+  const current = history[stepNumber];
+  const status = checkStatus();
+  const moves = history.map(movement);
+
+  const getClassWinner = () => {
```

Comment:
I'd convert this comp to a class component, and move this func to a instance of the class, for avoid create a new instance of this function in each render

Reply:
Perfect, on my way.